### PR TITLE
Cherry-pick LwM2M and CoAP fixes from upstream

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Fetch LwM2M and CoAP related bugfixes and improvements.

### CODE REVIEW

**Review by jheiskan81 — APPROVED** [review]

**Comment by SeppoTakalo** [comment]
Boot-fw-update tests fail constantly:
```
java.io.IOException: HTTP error code: 429
URI: http://ci-health:9200/jenkins-console/_doc
RESPONSE: HttpResponseProxy{HTTP/1.1 429 Too Many Requests 
```

**Comment by SeppoTakalo** [comment]
@jhirsi or @trantanen can you also approve the MoSH related part?

File: west.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 06088342cb531ee470c7f64c8c7c0e10d75de8fc
+      revision: pull/1325/head
```
Comment (rlubos) [inline west.yml:64]: Please update SHA: `fcbf3786cd8f40829dfe4cd3b1421c63437a47b6`

**Review by tokangas — APPROVED** [review]
Looks good from `modem_shell` point of view if the new event does not require any action.

**Comment by SeppoTakalo** [comment]
> Looks good from `modem_shell` point of view if the new event does not require any action.

No.
We just added it so we could react into it, for example if we have forced some modem state.. But by default, it does not need reaction.

File: west.yml (hunk 1 above)
Comment (SeppoTakalo) [reply west.yml]: Done

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
CI (mostly) passed on https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12498/5/, the only failure was on fw-update, where one of the jobs could not start.

**Pull request merged**